dependencies.go

```diff
@@ -27,19 +31,133 @@ type (
 	}
 )
 
+// dependencyCommitFail corrupts the first page of a transaction when it
+// is committed
+type dependencyCommitFail struct {
+	prodDependencies
+}
+
+func (dependencyCommitFail) disrupt(s string) bool {
+	if s == "CommitFail" {
+		return true
+	}
+	return false
+}
+
+// dependencyReleaseFail corrupts the first page of a transaction when it
+// is released
+type dependencyReleaseFail struct {
+	prodDependencies
+}
+
+func (dependencyReleaseFail) disrupt(s string) bool {
+	if s == "ReleaseFail" {
+		return true
+	}
+	return false
+}
+
 // prodDependencies is a passthrough to the standard library calls
 type prodDependencies struct{}
 
 func (prodDependencies) disrupt(string) bool { return false }
-
 func (prodDependencies) readFile(path string) ([]byte, error) {
 	return ioutil.ReadFile(path)
 }
-
 func (prodDependencies) openFile(path string, flag int, perm os.FileMode) (file, error) {
 	return os.OpenFile(path, flag, perm)
 }
-
 func (prodDependencies) create(path string) (file, error) {
 	return os.Create(path)
 }
+
+// faultyFile implements a file that simulates a faulty disk.
+type faultyFile struct {
```

Comment:
I feel like mocked dependencies belong in test files, but I could be convinced otherwise

Reply:
I actually like it that way. Dependencies feel like something that should be reusable in different test files and this way if you look for a dependency you know where to find the code. Another advantage I could think of is, that this prevents the test files from being unnecessarily cluttered.

Reply:
dependency_test.go

I agree that the mocked deps should all be in the same place so it's easy to reuse them. But that place should probably be dependecy_test.go.